### ISSUE TITLE
Ping probe crashes if a target doesn't resolve at all

### ISSUE THREAD

**Issue by VincentS** [google/cloudprober#264]
Configuring multiple Non-Datagram PING probes :

```
    probe {
      name: "xyz_git"
      type: PING
      targets {
        host_names: "git.xyz.net"
      }
      interval_msec: 5000  # 5s
      timeout_msec: 1000   # 1s
      ping_probe {
        use_datagram_socket: false
      }
    }
    probe {
      name: "xyz_nexus_server"
      type: PING
      targets {
        host_names: "nexus.xyz.net"
      }
      interval_msec: 5000  # 5s
      timeout_msec: 1000   # 1s
      ping_probe {
        use_datagram_socket: false
      }
    }

```

Causes an segmentation fault when running CloudProber:

```
 [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xbc2294]

goroutine 200 [running]:
github.com/google/cloudprober/probes/ping.(*Probe).Start(0xc0003f2320, 0xff51e0, 0xc00027e880, 0xc000127200)
	/home/travis/gopath/src/github.com/google/cloudprober/probes/ping/ping.go:466 +0x324
created by github.com/google/cloudprober/prober.(*Prober).startProbe
	/home/travis/gopath/src/github.com/google/cloudprober/prober/prober.go:268 +0x16f
```

**Comment by manugarg** [comment]
@VincentS Thanks for the bug report. I was not to able to reproduce it right away, so I had a few questions:

- Does it happen immediately after starting cloudprober, or randomly after some time?
- Which version of cloudprober are you running? (for stack trace and source code matching).

Anything else you can share about your setup.

**Comment by manugarg** [comment]
Also, was there another line above the segmentation violation code line in the error message?

**Comment by manugarg** [comment]
Ah, I was able to finally reproduce it. This bug gets triggered only if one of the targets doesn't get resolved. Basically whenever we take the following code path, "result" data  struct doesn't get created for a target:

https://github.com/google/cloudprober/blob/206e5fcb9193bcbfe07950ea49a04ab08f985c51/probes/ping/ping.go#L207

I'll send a fix soon.

Thanks again for reporting this bug.

**Comment by VincentS** [comment]
Thank you for the quick response / fix.

**Comment by manugarg** [comment]
Released the fixed binary in v0.10.4.
